src/language/lexer.ts

```diff
@@ -6,6 +6,18 @@ import { isDigit, isNameContinue, isNameStart } from './characterClasses.js';
 import type { Source } from './source.js';
 import { TokenKind } from './tokenKind.js';
 
+/**
+ * Configuration options to control lexer behavior
+ */
+export interface LexerOptions {
+  /**
+   * By default, ignored tokens are valid syntax and ignored when lexing.
+   * This may be disabled for certain grammars that specifically disallow
+   * ignored tokens (e.g. schema coordinates).
+   */
+  noIgnoredTokens?: boolean | undefined;
```

Comment:
Just a "naming things" thing, but should we call it "forbid" or similar instead, to make the meaning clearer?

```suggestion
  forbidIgnoredTokens?: boolean | undefined;
```